File: S1-Matlab_AudioProcessor/audio_processor.m
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Offline Audio Processor %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%



%% Parameters

% This must be an audio file readable by Matlab audioread function
inputFile = 'Marimba.wav';
outputFile = 'output.wav';



%% Read input file

[inputData, sampleRate] = audioread(inputFile);
inputData = mean(inputData,2); % downmix to mono if neede
inputTime = (0:length(inputData)-1)/sampleRate;

figure(1);
plot(inputTime, inputData);
grid on;
title('Input audio data');
xlabel('time (seconds)');
ylabel('amplitude');

soundsc(inputData,sampleRate);
pause(length(inputData)/sampleRate);



%% Audio data process

outputData = zeros(size(inputData));

%%%%%%%%%%%%%%%%%%%%%%%%%
% PROCESSING COMES HERE %
%%%%%%%%%%%%%%%%%%%%%%%%%

outputData = inputData;

%%%%%%%%%%%%%%%%%%%%%%%%%



%% Save output

audiowrite(outputFile,outputData, sampleRate);
soundsc(outputData,sampleRate);
figure(2);
outputTime = (0:length(outputData)-1)/sampleRate;
plot(outputTime, outputData);
grid on;
title('Output audio data');
xlabel('time (seconds)');
ylabel('amplitude');

File: S1-Python_AudioProcesor/audio_processor.py
import numpy as np
import scipy.io.wavfile as wavfile


# Parameters
# **********

# This must be an audio file readable by scipy.io.wavfile.read() function
filename = 'Marimba.wav'


# Read input file
# ***************

fs, input_data = wavfile.read(filename, mmap=True)
# convert to float
max_value = float(-np.iinfo(input_data.dtype).min)
input_data = input_data.astype('float32') / max_value
# convert to mono if not already
if len(input_data.shape) > 1:
    input_data = np.mean(input_data, axis=1)
input_data = input_data.flatten()


# Audio data process
# ******************

output_data = np.zeros(len(input_data))

# *************************
# * PROCESSING COMES HERE *
# *************************

output_data = input_data

# *************************


# Save output
# ***********

wavfile.write('output.wav', int(fs), output_data)